uefi_test.go

```diff
@@ -43,11 +44,57 @@ func TestAppendUEFIFirmwareDeviceConfig(t *testing.T) {
 		t.Errorf("Expected non-empty qemuParams, found %s", c.qemuParams)
 	}
 
-	expected := "-drive if=pflash,format=raw,readonly,file=OVMF_CODE.fd -drive if=pflash,format=raw,file=OVMF_VARS.fd"
+	expected := "-drive if=pflash,format=raw,readonly=on,file=OVMF_CODE.fd -drive if=pflash,format=raw,file=OVMF_VARS.fd"
 	result := strings.Join(c.qemuParams, " ")
 	if expected != result {
 		t.Fatalf("Failed to append parameters\nexpected[%s]\n!=\nfound   [%s]", expected, result)
 	}
 }
 
+func TestAppendUEFIFirmwareDeviceArch(t *testing.T) {
+	udev := UEFIFirmwareDevice{Code: "AAVMF_CODE.ms.fd", Vars: "AAVMF_VARS.ms.fd"}
+	expected := "-drive if=pflash,format=raw,readonly=on,file=AAVMF_CODE.ms.fd -drive if=pflash,format=raw,file=AAVMF_VARS.ms.fd"
+	testAppend(udev, expected, t)
+}
+
+func TestAppendUEFIFirmwareDeviceConfigArch(t *testing.T) {
+	c := &Config{}
+	udev := UEFIFirmwareDevice{Code: "AAVMF_CODE.ms.fd", Vars: "AAVMF_VARS.ms.fd"}
+	c.UEFIFirmwareDevices = append(c.UEFIFirmwareDevices, udev)
+	err := c.appendDevices()
+	if err != nil {
+		t.Fatalf("Failed to append UEFI firwmware device: %s", err)
+	}
+	if len(c.qemuParams) == 0 {
+		t.Errorf("Expected non-empty qemuParams, found %s", c.qemuParams)
+	}
+
+	expected := "-drive if=pflash,format=raw,readonly=on,file=AAVMF_CODE.ms.fd -drive if=pflash,format=raw,file=AAVMF_VARS.ms.fd"
+	result := strings.Join(c.qemuParams, " ")
+	if expected != result {
+		t.Fatalf("Failed to append parameters\nexpected[%s]\n!=\nfound   [%s]", expected, result)
+	}
+}
+
+func TestNewFirmwareDev(t *testing.T) {
+	udev_ptr, err := NewSystemUEFIFirmwareDevice(true)
+	if err != nil {
+		t.Fatalf("Failed to find secure firmware blobs: %s", err)
+	}
+	udev := *udev_ptr
+	switch runtime.GOARCH{
+	case "s390x", "amd64":
+		if PathExists(UbuntuSecVars) {
+			expected := "-drive if=pflash,format=raw,readonly=on,file=/usr/share/OVMF/OVMF_CODE.secboot.fd -drive if=pflash,format=raw,file=/usr/share/OVMF/OVMF_VARS.ms.fd"
+			testAppend(udev, expected, t)
+		} else if PathExists(CentosSecVars){
+			expected := "-drive if=pflash,format=raw,readonly=on,file=/usr/share/OVMF/OVMF_CODE.secboot.fd -drive if=pflash,format=raw,file=/usr/share/OVMF/OVMF_VARS.secboot.fd"
+			testAppend(udev, expected, t)
+		}
```

Comment:
If neither exist, then we won't report any failure; we should.

at the end of the if/else-if, you can just do a `t.Fatal("failed to find secure firmware blogs")`